### PR TITLE
jasper: add v4.2.4, drop vulnerable versions (< 4.1.2)

### PR DESCRIPTION
**Pull request by valgur**
Based on https://repology.org/project/jasper/information

* Adding v4.2.4
* Keeping 4.0.0:
```
gdal/pre_3.5.0/conanfile.py
261:            self.requires("jasper/4.0.0")

opencv/3.x/conanfile.py
89:            self.requires("jasper/4.0.0")

opencv/2.x/conanfile.py
373:            self.requires("jasper/4.0.0")

libraw/all/conanfile.py
61:            self.requires("jasper/4.0.0")
```

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by franramirez688 — CHANGES_REQUESTED** [review]
Hi @valgur - Thanks for the PR!

I can see all these libs still using `jasper/4.0.0`:

```
gdal/pre_3.5.0/conanfile.py
261:            self.requires("jasper/4.0.0")

opencv/3.x/conanfile.py
89:            self.requires("jasper/4.0.0")

opencv/2.x/conanfile.py
373:            self.requires("jasper/4.0.0")

libraw/all/conanfile.py
61:            self.requires("jasper/4.0.0")
```

So, let's keep that version for now.

File: recipes/jasper/all/conandata.yml [hunk 1]
```diff
@@ -1,72 +1,27 @@
 sources:
+  "4.2.2":
+    url: "https://github.com/jasper-software/jasper/releases/download/version-4.2.2/jasper-4.2.2.tar.gz"
+    sha256: "5e397570b3110a8edef6d50127e20a2297939809cea25d29068823b442ecdd6d"
```
Comment (franramirez688) [inline recipes/jasper/all/conandata.yml:4]:
```suggestion
  "4.2.4":
    url: "https://github.com/jasper-software/jasper/releases/download/version-4.2.4/jasper-4.2.4.tar.gz"
    sha256: "6a597613d8d84c500b5b83bf0eec06cd3707c23d19957f70354ac2394c9914e7"
```

File: recipes/jasper/all/conandata.yml [hunk 2]
```diff
@@ -1,72 +1,27 @@
...
-    sha256: "39514e1b53a5333fcff817e19565371f016ea536c36fd2d13a9c4d8da8f0be0c"
-  "3.0.6":
-    url: "https://github.com/jasper-software/jasper/releases/download/version-3.0.6/jasper-3.0.6.tar.gz"
-    sha256: "169be004d91f6940c649a4f854ada2755d4f35f62b0555ce9e1219c778cffc09"
-  "2.0.33":
-    url: "https://github.com/jasper-software/jasper/releases/download/version-2.0.33/jasper-2.0.33.tar.gz"
-    sha256: "28d28290cc2eaf70c8756d391ed8bcc8ab809a895b9a67ea6e89da23a611801a"
 patches:
-  "4.2.0":
+  "4.2.2":
```
Comment (franramirez688) [inline recipes/jasper/all/conandata.yml:12]:
```suggestion
  "4.2.4":
```

File: recipes/jasper/config.yml [hunk 3]
```diff
@@ -1,15 +1,7 @@
 versions:
+  "4.2.2":
```
Comment (franramirez688) [inline recipes/jasper/config.yml:2]:
```suggestion
  "4.2.4":
```

**Review by franramirez688 — COMMENTED** [review]
I just saw the version 4.2.4 released 😁

**Review by franramirez688 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM. Only versions 4.2.0 and 2.0.0 are used in CCI.

**Pull request merged**